osx/KA-Lite/KA-Lite/AppDelegate.m

```diff
@@ -604,12 +605,46 @@ - (IBAction)discardPreferences:(id)sender {
     [self discardPreferences];
 }
 
+- (IBAction)kaliteUninstall:(id)sender {
+        
+    // Get the KA Lite application directory path.
+    NSString *appPath = [[NSBundle mainBundle] bundlePath];
+    NSRange searchLastWord = [appPath rangeOfString:@"/" options:NSBackwardsSearch];
+    NSString *lastWord = [appPath substringFromIndex:searchLastWord.location+1];
+    NSString *kaliteAppDir = [appPath stringByReplacingOccurrencesOfString:lastWord withString:@""];
+    NSString *kaliteUninstallPath = [NSString stringWithFormat: @"%@%@", kaliteAppDir, @"KA-Lite_Uninstall.tool"];
+        
+    if (pathExists(kaliteUninstallPath)) {
+    
+        if (confirm(@"Are you sure that you want to uninstall the KA Lite application?")) {
+            
+            const char *runCommand;
+            if ([self.deleteKaliteData state]==NSOnState) {
+                // Delete the KA Lite data.
+                NSString *kaliteUninstallData = [NSString stringWithFormat: @"%@ %@", kaliteUninstallPath, @"yes yes"];
+                runCommand = [kaliteUninstallData UTF8String];
+            } else {
+                NSString *kaliteAppUninstall = [NSString stringWithFormat: @"%@ %@", kaliteUninstallPath, @"yes no"];
+                runCommand = [kaliteAppUninstall UTF8String];
+            }
+            
+            int runCommandStatus = system(runCommand);
+            
+            if (runCommandStatus == 0) {
+                // Terminate application.
+                [[NSApplication sharedApplication] terminate:nil];
+            }
```

Comment:
Can you show an alert message that the uninstall did not succeed?  Would also be helpful to mention that the user may look at the logs (or the Console application) for more information.